zyte_spider_templates/heuristics.py

```diff
@@ -56,3 +65,44 @@ def might_be_category(url: str) -> bool:
                 return False
 
     return True
+
+
+INDEX_URL_PATHS = {
+    "",
+    "/index",
+    "/index.html",
+    "/index.htm",
+    "/index.php",
+    "/home",
+}
+
+
+def is_homepage(url: str) -> bool:
```

Comment:
hey @BurnzZ!

I've run an experiment:

1. Got a list of popular and less popular domains (~160K)
2. Created a spider, which just downloads the front page, following all redirects.
3. Then, if after all redirects the URL is not barebone `/`, stored created an item.

So, this way we can get a dataset of positive homepages. It's a bit simplistic - there could be homepage variants with different URLs, and it won't help with catching false positives (that's a separate question, how to get a dataset for this).

In the end, I've got ~10K example non-trivial homepage URLs: https://gist.github.com/kmike/bcc970549e428ee5518868988ebd39dd

Could you please check the heuristics against this list? 

As currently we don't have a dataset to catch false positives, we need to be careful and conservative with adding more heuristics, but it seems there could be some common patterns which can be added to improve the results

Reply:
So far, the `is_homepage()` heuristic detects around 50% of the URLs as homepages. It's usually the case of having these suffix patterns in the URL:

- `/de/`
- `/en`
- `/home`
- `/en-US`
- `/en-us`
- `/index.php`
- `/us/en`
- `/pt_br/index`
- `/it_IT`
- `/en/home/`
- `/en/index.html?`

The other 50% of the URLs which were deemed to not be URLs go something like:

- `/content/boaml/en_us/home.html`
- `/DIGITAL/Navigation/DE/Home/home.html`
- `/all-about/oxfordshire`
- `/en/services/health.html`
- `/forums/index.php`
- `?gws_rd=ssl` - or basically any URL with Query params 
- `/en/games`
- `/portal`
- `/archives/main/`
- `/pages/homepage/`
- `/Pages/Default.aspx`
- `/en/page/home`

So far, I think the current setup of the `is_homepage()` heuristics is conservative enough. Most of them were False Negatives as can be observed from 50% of the results not detected as homepages.